modules/building-memcached-operator-using-osdk.adoc

```diff
@@ -0,0 +1,158 @@
+
+.. Watch for the Memcached custom resource definition (CRD). By default, the `memcached-operator` watches `Memcached` resource events as shown in `cmd/memcached-operator/main.go`:
++
+----
+func main() {
+	printVersion()
+
+	sdk.ExposeMetricsPort()
+
+	resource := "cache.example.com/v1alpha1"
+	kind := "Memcached"
+	namespace, err := k8sutil.GetWatchNamespace()
+	if err != nil {
+		logrus.Fatalf("Failed to get watch namespace: %v", err)
+	}
+	resyncPeriod := 5
+	logrus.Infof("Watching %s, %s, %s, %d", resource, kind, namespace, resyncPeriod)
+	sdk.Watch(resource, kind, namespace, resyncPeriod)
+	sdk.Handle(stub.NewHandler())
+	sdk.Run(context.TODO())
+}
+----
+
+.. Define the Memcached spec and status.
+
+... Modify the spec and status of the `Memcached` CRD at
+`pkg/apis/cache/v1alpha1/types.go`:
++
+----
+type MemcachedSpec struct {
+	// Size is the size of the memcached deployment
+	Size int32 `json:"size"`
+}
+type MemcachedStatus struct {
+	// Nodes are the names of the memcached pods
+	Nodes []string `json:"nodes"`
+}
+----
+
+... Update the generated code for the CR:
++
+----
+$ cd $GOPATH/src/github.com/example-inc/
+$ cd memcached-operator
+$ operator-sdk generate k8s
+----
+
+.. Define the Handler. The reconciliation loop for an event is defined in the `Handle()` function at `pkg/stub/handler.go`.
++
+Replace this file with the reference implementation found at
+link:https://github.com/operator-framework/getting-started/blob/master/handler.go.tmpl#L7[handler_go].
+You must update the highlighted line if you have changed the import path of this
+project to something other than `example-inc`.
++
+[NOTE]
+====
+The provided handler implementation is only meant to demonstrate the use of the
+SDK APIs and is not representative of the best practices of a reconciliation
+loop.
+====
+
+. Build and run the Operator.
+
+.. Build the `memcached-operator` image and push it to a registry. Ensure you have
+an account on link:https://quay.io[quay.io] for the next step, or substitute
+your preferred container registry. On the registry,
+link:https://quay.io/new/[create a new public image] repository named
+`memcached-operator`.
++
+----
+$ cd $GOPATH/src/github.com/example-inc/
+$ cd memcached-operator
+$ operator-sdk build quay.io/example/memcached-operator:v0.0.1
+$ docker push quay.io/example/memcached-operator:v0.0.1
+----
++
+Kubernetes deployment manifests are generated in the *_deploy/operator.yaml_*
+file. The deployment image is set to the container image specified above.
+
+.. Deploy the Memcached Operator:
++
+----
+$ kubectl create -f deploy/rbac.yaml
+$ kubectl create -f deploy/crd.yaml
+$ kubectl create -f deploy/operator.yaml
+----
+
+.. Verify that `memcached-operator` is up and running:
++
+----
+$ kubectl get pods
+NAME                                  READY     STATUS    RESTARTS   AGE
+memcached-operator-75c4b4c665-8jnj5   1/1       Running   0          20s
+----
+
+.. After confirming the pod is running, clean everything up:
```

Comment:
Looking at this with fresh eyes, we are missing the step where you actually deploy a Memcache object and see the Operator do something with it. This part of it: https://github.com/operator-framework/operator-sdk/blob/master/doc/user-guide.md#create-a-memcached-cr

Reply:
@robszumski OK, added those!

http://file.rdu.redhat.com/~adellape/083018/osdk_gs/operators/osdk-getting-started.html#building-memcached-oeprator-using-osdk_osdk-getting-started